src/corehost/cli/deps_resolver.cpp

```diff
@@ -403,6 +417,75 @@ void deps_resolver_t::init_known_entry_path(const deps_entry_t& entry, const pal
     }
 }
 
+void deps_resolver_t::resolve_additional_deps(const hostpolicy_init_t& init)
+{
+    pal::string_t additional_deps_serialized = init.additional_deps_serialized;
+    pal::string_t fx_name = init.fx_name;
+    pal::string_t fx_ver = init.fx_ver;
+
+    if (additional_deps_serialized.empty())
+    {
+        return;
+    }
+
+    pal::string_t additional_deps_path;
+    pal::stringstream_t ss(additional_deps_serialized);
+
+    // Process the delimiter separated custom deps files.
+    while (std::getline(ss, additional_deps_path, PATH_SEPARATOR))
+    {
+        // If it's a single deps file, insert it in 'm_additional_deps_files'
+        if (ends_with(additional_deps_path, _X(".deps.json"), false))
+        {
+            if (pal::file_exists(additional_deps_path))
+            {
+                m_additional_deps_files.push_back(additional_deps_path);
+            }
+        }
+        // If it's not a single deps file, we'll assume it's a directory
+        // and process it only for portable apps since they carry
+        // FX name and version. 
+        // 
+        // Standalone apps do not carry that information but are published
+        // incontext of a FX/Ver/RID. Thus, explicit path to deps.json
+        // must be specified for them.
+        else if (m_portable)
+        {
+            // We'll search deps files in 'base_dir'/shared/fx_name/fx_ver
+            append_path(&additional_deps_path, _X("shared"));
+            append_path(&additional_deps_path, fx_name.c_str());
+            append_path(&additional_deps_path, fx_ver.c_str());
+
+            // The resulting list will be empty if 'additional_deps_path' is not a valid directory path
+            std::vector<pal::string_t> list;
+            pal::readdir(additional_deps_path, _X("*.deps.json"), &list);
+            for (pal::string_t json_file : list)
+            {
+                pal::string_t json_full_path = additional_deps_path;
+                append_path(&json_full_path, json_file.c_str());
+                m_additional_deps_files.push_back(json_full_path);
+            }
+        }
+    }
+
+    if (m_portable)
+    {
+        for (pal::string_t json_file : m_additional_deps_files)
+        {
+            m_additional_deps.push_back(std::unique_ptr<deps_json_t>(
+                new deps_json_t(true, json_file, m_fx_deps->get_rid_fallback_graph())));
```

Comment:
Oh right.  Now I remember, in the app.deps.json, it has the asset path:

```
        "native": {
          "runtimes/win7-x64/native/libuv.dll": {}
        }
```

Which is how the host knows how to load it.  The "fallback" logic happens in NuGet, not at runtime.